File: hash/leetcode_73/solution.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        row_group = set()
        col_group = set()
        
        row_len = len(matrix)
        col_len = len(matrix[0])

        for i in range(row_len):
            for j in range(col_len):
                if matrix[i][j] == 0:
                    row_group.add(i)
                    col_group.add(j)

        for row in row_group:
            matrix[row] = [0] * col_len
        
        for col in col_group:
            for i in range(row_len):
                matrix[i][col] = 0

File: hash/leetcode_73/solution.js
/**
 * @param {number[][]} matrix
 * @return {void} Do not return anything, modify matrix in-place instead.
 */
var setZeroes = function(matrix) {
    const rowGroup = new Set();
    const colGroup = new Set();

    const rowLength = matrix.length;
    const colLength = matrix[0].length;

    for(let i=0; i<rowLength; i++) {
        for(let j=0; j<colLength; j++) {
            if(matrix[i][j] === 0) {
                rowGroup.add(i);
                colGroup.add(j)
            }
        }
    }

    for(const row of rowGroup) {
        for(let i=0; i<colLength; i++) {
            matrix[row][i] = 0;
        }
    }

    for(const col of colGroup) {
        for(let i=0; i<rowLength; i++) {
            matrix[i][col] = 0;
        }
    }
};